### PR TITLE
issue/#3027/Books counts are not shown properly

### PR DESCRIPTION
**Pull request by Ranjeet-Kalsi**
Targets #3027 

### Description
I have found the total numbers of books shown in the Wishlist's menu & Borrowed Books menu.

I have added to lines in    resources\views\knowledgecafe\library\menu.blade.php

  $wishlistCount = auth()->user()->booksInWishlist->count();
 $borrowedCount = auth()->user()->booksBorrower->count();

### Checklist:
First i checked the flow of code then apply changes to menu.blade.php
books
wishlist
borrowed

### CODE REVIEW

File: resources/views/knowledgecafe/library/menu.blade.php [hunk 1]
```diff
@@ -23,13 +23,15 @@
     <li class="nav-item">
         @php
          $params = array_merge($request,  ['wishlist' => 'booksInWishlist']);
+         $wishlistCount = auth()->user()->booksInWishlist->count();
```
Comment (Ayush8923) [inline resources/views/knowledgecafe/library/menu.blade.php:26]: Please write this logic in service then return the `$wishlistCount` variable from there and directly use it here.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Added one comment. Please check.

File: resources/views/knowledgecafe/library/menu.blade.php (hunk 1 above)
Comment (Ranjeet-Kalsi) [reply resources/views/knowledgecafe/library/menu.blade.php:26]: ok I will check it

**Comment by LavishRawat** [comment]
@Ranjeet-Kalsi The ``Books Category``  and  ``A Book A Month``  pages are currently displaying the following error.
Please have a look and let me know if it's the same for you.

Screenshot 2023-06-21 at 10 37 52 AM

File: resources/views/knowledgecafe/library/menu.blade.php [hunk 2]
```diff
@@ -6,6 +6,8 @@
      $borrowedSelected = request()->input('borrowedBook','active') === 'markedAsBorrowed';
 
      $allBooksSelected =  $active === 'books' && ! $wishlistSelected && ! $borrowedSelected;
+     $wishlistedBooksCount = auth()->user()->booksInWishlist->count();
+     $booksBorrowerCount = auth()->user()->booksBorrower->count();
```
Comment (Ayush8923) [inline resources/views/knowledgecafe/library/menu.blade.php:10]: Now, these 2 lines of code are not needed here. Please remove this. And then please test locally once.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Added one small comment.

**Comment by Ayush8923** [comment]
@Ranjeet-Kalsi Suggested changes are not implemented. Could you please check this [comment](https://github.com/ColoredCow/portal/pull/3172#pullrequestreview-1489846697)?
And please test locally once and also please do self-review before assigning the PR to reviewers.

**Comment by Ranjeet-Kalsi** [comment]
> @Ranjeet-Kalsi Suggested changes are not implemented. Could you please check this [comment](https://github.com/ColoredCow/portal/pull/3172#pullrequestreview-1489846697)? And please test locally once and also please do a self-review before assigning the PR to reviewers.

yes i not implemented the suggested changes I used only in locally.
ok i will implemented it

**Comment by Ranjeet-Kalsi** [comment]
> @Ranjeet-Kalsi Suggested changes are not implemented. Could you please check this [comment](https://github.com/ColoredCow/portal/pull/3172#pullrequestreview-1489846697)? And please test locally once and also please do a self-review before assigning the PR to reviewers.

Now suggested changes applied and also i follow the these steps 
1. remove count logic from controller file due to another function affects 
2. i implemented code & logic only in blade file

File: resources/views/knowledgecafe/library/menu.blade.php [hunk 3]
```diff
@@ -6,8 +6,11 @@
      $borrowedSelected = request()->input('borrowedBook','active') === 'markedAsBorrowed';
 
      $allBooksSelected =  $active === 'books' && ! $wishlistSelected && ! $borrowedSelected;
+     $wishlistedBooksCount = auth()->user()->booksInWishlist->count();
+     $booksBorrowedCount = auth()->user()->booksBorrower->count();
```
Comment (Ayush8923) [inline resources/views/knowledgecafe/library/menu.blade.php:10]: @Ranjeet-Kalsi, you don't need to add these books count logic inside the blade file. let's define these 2 books count logic inside the  `index` function of the `BookController` and `index` function of the `BookCategoryController` and then return these 2 variable corresponding to their `index.blade` file. and after that, we can directly use this variable in the blade file. 

Please let me know, if there are any confusion and question.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Added one comment.

File: app/Http/Controllers/KnowledgeCafe/Library/BookCategoryController.php [hunk 4]
```diff
@@ -23,10 +23,14 @@ public function index()
...
         $books = Book::all();
+        $wishlistedBooksCount = auth()->user()->booksInWishlist->count();
+        $booksBorrowedCount = auth()->user()->booksBorrower->count();
 
         return view('knowledgecafe.library.categories.index')
         ->with('books', $books)
-        ->with('categories', $this->formatCategoryData($categories));
+        ->with('categories', $this->formatCategoryData($categories))
+        ->with('wishlistedBooksCount', $wishlistedBooksCount)
+        ->with('booksBorrowedCount', $booksBorrowedCount);
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookCategoryController.php:33]: I think we can optimize this code. Something like this.
```
return view('knowledgecafe.library.categories.index', [
    'books' => $books,
    'categories' => $this->formatCategoryData($categories),
    'wishlistedBooksCount' => $wishlistedBooksCount,
    'booksBorrowedCount' => $booksBorrowedCount,
]);
```

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php [hunk 5]
```diff
@@ -328,7 +330,13 @@ public function bookAMonthIndex()
...
-
-        return view('knowledgecafe.library.books.book-a-month')->with('booksCollection', $booksCollection)->with('books', $books);
+        $wishlistedBooksCount = auth()->user()->booksInWishlist->count();
+        $booksBorrowedCount = auth()->user()->booksBorrower->count();
+
+        return view('knowledgecafe.library.books.book-a-month')
+        ->with('booksCollection', $booksCollection)
+        ->with('books', $books)
+        ->with('wishlistedBooksCount', $wishlistedBooksCount)
+        ->with('booksBorrowedCount', $booksBorrowedCount);
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookController.php:340]: Same here. Could you please optimize this code?
 Please don't repeat the `with` function again and again. Please take the reference of the previous comment.

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Added some comments. Please check.

File: app/Http/Controllers/KnowledgeCafe/Library/BookCategoryController.php [hunk 6]
```diff
@@ -23,10 +23,14 @@ public function index()
...
+        $booksBorrowedCount = auth()->user()->booksBorrower->count();
 
-        return view('knowledgecafe.library.categories.index')
-        ->with('books', $books)
-        ->with('categories', $this->formatCategoryData($categories));
+        return view('knowledgecafe.library.categories.index', [
+        'books'=> $books,
+        'categories'=> $this->formatCategoryData($categories),
+        'wishlistedBooksCount'=> $wishlistedBooksCount,
+        'booksBorrowedCount'=> $booksBorrowedCount]);
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookCategoryController.php:33]: Please fix the indentation here.

File: app/Http/Controllers/KnowledgeCafe/Library/BookController.php [hunk 7]
```diff
@@ -328,7 +330,13 @@ public function bookAMonthIndex()
...
-
-        return view('knowledgecafe.library.books.book-a-month')->with('booksCollection', $booksCollection)->with('books', $books);
+        $wishlistedBooksCount = auth()->user()->booksInWishlist->count();
+        $booksBorrowedCount = auth()->user()->booksBorrower->count();
+
+        return view('knowledgecafe.library.books.book-a-month', [
+        'booksCollection'=> $booksCollection,
+        'books'=> $books,
+        'wishlistedBooksCount'=> $wishlistedBooksCount,
+        'booksBorrowedCount'=> $booksBorrowedCount]);
```
Comment (Ayush8923) [inline app/Http/Controllers/KnowledgeCafe/Library/BookController.php:340]: Same here. Please fix the indentation here..

**Review by Ayush8923 — CHANGES_REQUESTED** [review]
Few minor comments. Please check.

**Review by Ayush8923 — APPROVED** [review]

**Pull request merged**